### ISSUE TITLE
Use other iso than boot2docker to create machine based on virtualbox, but failed

### ISSUE THREAD

**Issue by TrumanLing** [docker/machine#669]
Input to create machine:
```console
truman@truman:~/work/machine-bak$ sudo ./docker-machine_linux-amd64 -D create -d virtualbox --virtualbox-boot2docker-url=http://rnd-isourceb.huawei.com/docker/dockos/blob/master/DockOS.iso dev
```
output:
```console
DEBU[0000] STDERR:                                      
INFO[0000] Downloading boot2docker.iso from http://rnd-isourceb.huawei.com/docker/dockos/blob/master/DockOS.iso... 
ERRO[0039] Error creating machine: open /home/truman/.docker/machine/cache/boot2docker.iso/boot2docker.iso.tmp108446320: not a directory
```

**Comment by shawnzhu** [comment]
The error message indicated that the downloading path is wrong. see [my comment](/docker/machine/issues/670#issuecomment-77972562)

**Comment by shawnzhu** [comment]
@TrumanLing the option `--virtualbox-boot2docker-url` is fixed in #805
